ValveResourceFormat/Utils/StreamHelpers.cs

```diff
@@ -20,26 +20,20 @@ public static string ReadNullTermString(this BinaryReader stream, Encoding encod
             }
 
             var characterSize = encoding.GetByteCount("e");
+            Span<byte> data = stackalloc byte[characterSize];
 
             using var ms = new MemoryStream();
 
             while (true)
             {
-                var data = new byte[characterSize];
+                var bytesRead = stream.Read(data);
 
-                int bytesRead;
-                var totalRead = 0;
-                while ((bytesRead = stream.Read(data, totalRead, characterSize - totalRead)) != 0)
-                {
-                    totalRead += bytesRead;
-                }
-
-                if (encoding.GetString(data, 0, characterSize) == "\0")
+                if (bytesRead == 0 || encoding.GetString(data) == "\0")
```

Comment:
Hm. If `bytesRead` is less than `characterSize`, data will have garbage value. For example it might not encode as NULL with UTF16.